core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -959,6 +960,54 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     partIdFromPartKeyLookupLatency.record(System.nanoTime - startExecute)
     chosenPartId
   }
+
+  /**
+   * Iterate through the LuceneIndex and calculate cardinality count
+   */
+  def calculateCardinality(partSchema: PartitionSchema, cardTracker: CardinalityTracker): Unit = {
+    val coll = new CardinalityCountBuilder(partSchema, cardTracker)
+    withNewSearcher(s => s.search(new MatchAllDocsQuery(), coll))
+    // IMPORTANT: making sure to flush all the data in rocksDB
+    cardTracker.flushCardinalityCount()
+  }
+}
+
+/**
+ * In this lucene index collector, we read through the entire lucene index periodically and re-calculate
+ * the cardinality count from scratch. This class iterates through each document in lucene, extracts a shard-key
+ * and updates the cardinality count using the given CardinalityTracker.
+ * */
+class CardinalityCountBuilder(partSchema: PartitionSchema, cardTracker: CardinalityTracker) extends SimpleCollector {
+
+  private var partKeyDv: BinaryDocValues = _
+  private var singleResult: BytesRef = _
+
+  // gets called for each segment
+  override def doSetNextReader(context: LeafReaderContext): Unit = {
+    partKeyDv = context.reader().getBinaryDocValues(PartKeyLuceneIndex.PART_KEY)
+  }
+
+  // gets called for each matching document in current segment
+  override def collect(doc: Int): Unit = {
+    if (partKeyDv.advanceExact(doc)) {
+      singleResult = partKeyDv.binaryValue()
+      val localBytes = singleResult.bytes
+      val localMap = partSchema.binSchema.toStringPairs(localBytes, UnsafeUtils.arayOffset).toMap
+
+      // TODO: quick sync with team on correct way to get key names from config or from any other constants ??
```

Comment:
@amolnayak311 @sherali42 Wanted to quickly sync on the right way to get the workspace, namespace and metric's key name ? 

Also wondering if I need any additional checks here ?